/**
 * Copyright (c) 2022-present, Ukama Inc.
 * All rights reserved.
 *
 * This source code is licensed under the XXX-style license found in the
 * LICENSE file in the root directory of this source tree. An additional grant
 * of patent rights can be found in the PATENTS file in the same directory.
 */

syntax = "proto3";

package events;

// Event listener definition
service EventListener {
  // Send registration event
  rpc SystemRegistrationEvent(RegistrationRequest) returns (Response) {}
}

// Registration event.
message RegistrationRequest {

  string name = 1; // Name of the system, e.g., init
  string ip   = 2; // IP address where service is listening
  string port = 3; // Port number where service is binding 
}

// Event response.
message Response {
  string code = 1; // Success response code (200 OK, 500 Error)
}
